database/src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -145,6 +175,30 @@ public boolean removeRecordFromTable(Class<? extends BaseEntity> entityClass, In
         return true;
     }
 
+    @Override
+    public boolean removeSpecificRecords(Class<? extends BaseEntity> entityClass, List<Integer> ids) {
+        if (ids == null || ids.isEmpty()) {
+            throw new IllegalArgumentException("IDs list cannot be null or empty");
+        }
+
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        if (!Files.exists(databasePath)) {
```

Comment:
I see that this validation is not used in some methods, please add it into all methods where it can be useful.